docs/Architecture Decision Record/009-git-workflow.md

```diff
@@ -0,0 +1,93 @@
+
+Throughout the project all vendor development work has been done in the vendor's forked repository, while pull requests from the government (OFA and 18F) and even some documentation pull requests from the vendor were made directly to the government repository.
+
+This has created a situation where the vendor has needed to continuously rebase with the government repository to make sure the vendor's repository was up to date. As a result, problems with the Git history have arisen that make it confusing for both the government and the vendor to track the history of the work.
+
+The proposed workflow below provides a remedy to these issues, as well as many others, as detailed in the Consequences section.
+
+## Decision
+
+A contributor to the TDP project would always use the following steps to propose changes to the repository. No merges directly into `HHS:main`
+
+![](images/TANF-Git-Workflow.png)
+
+1. Check out `raft-tdp-main` in `raft-tech/TANF-app` (or another branch if this work is dependent on a branch that hasn't been merged yet)
+2. If working locally, run `git pull`
+3. Check out new feature branch `git checkout -b <BRANCH NAME>`
+4. Update the repo with code, documentation, etc.
+5. If working locally, run `git push origin <BRANCH NAME>`
+6. Create [a draft pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) from new branch to `raft-tdp-main`. The PR title should include WIP e.g. `WIP: Adding a feature` (If your work is dependent on another branch that has yet to be merged, issue the pull request against that branch)
+8. Implement, Test, Review work independently
+9. When finished: 
+    * Update the Pull Request Template
+    * Add in-line comment to the file changes to provide context for the proposed changes
+    * Ensure there are no merge conflicts 
+    * Ensure CI/CD pipelines are green
+    * Update the title to remove `WIP`, change the PR to [Ready for Review](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request), assign label `raft-review`.
+    * If your PR is against a branch other than `raft-tdp-main` keep it as a draft PR until the other branch is merged. 
+13. Assign a reviewer: 
+    * For development work, assign **at least two** Raft developers with one of them being `carltonsmith` or `jtwillis92` (**but not both**).
+    * For research and design assign `shubhi-raft` and `jenewingpierce`
+    * For security controls assign `carltonsmith`
+    * For documentation submitted by OFA and 18F tag `lfrohlich` and `adpennington` and remove label `raft-review`
+    * For documentation submissions and updates by raft assign `lfrohlich` and `adpennington`
```

Comment:
I don't quite understand this - 43 and 44 lines. Is this something @kniz-raft should be part of?

Reply:
That would make sense. I was under the impression that documentation changes usually went straight to gov review, we could definitely change that part of the flow to ask Kati to review it

Reply:
I've been making a lot of document submissions myself (sprint summaries, updates to how we work, etc) so not sure if I should be the reviewer if I'm also the contributor.

Reply:
The system won't allow you to approve your own pull request. I would imagine in that case you could ask @jenewingpierce or @shubhi-raft

Reply:
Got it. So to clarify replace 
>For documentation submissions and updates by raft assign `lfrohlich` and `adpennington`

with 
> For documentation submissions and updates by raft assign `lfrohlich` and `adpennington` and add QASP review label

Reply:
This is done